File: leetcode/349. Intersection of Two Arrays/intersection.ts
function intersection(nums1: number[], nums2: number[]): number[] {
    let num1: Set<number> = new Set<number>(nums1);
    let num2: Set<number> = new Set<number>(nums2);

    let arr: number[] = [];

    for (var a of num1) {
    for (var b of num2) {
        if (a==b) {
            arr.push(a);
            break;
            }
        }
    }
    return arr;
};

File: leetcode/349. Intersection of Two Arrays/intersection.cs
public class Solution {
    public int[] Intersection(int[] nums1, int[] nums2) {
        HashSet<int> num1 = new HashSet<int>(nums1);
        HashSet<int> num2 = new HashSet<int>(nums2);

        List <int> list = new List<int>();
        foreach (int a in num1) {
            foreach (int b in num2) {
                if (a == b) {
                    list.Add(a);
                    break;
                } 
            }
        }
        return list.ToArray();
    }
}
